put_option.go

```diff
@@ -0,0 +1,28 @@
+package statful
+
+type Put struct {
+	User string
+}
+
+type PutOption func(*Put)
+
+func NewPut(opts []PutOption) *Put {
+
+	const defaultUser = ""
+
+	p := &Put{
+		User: defaultUser,
+	}
+
+	for _, opt := range opts {
+		opt(p)
+	}
+
+	return p
+}
```

Comment:
Avoid exposing things as part of the public surface of the package whenever possible. In this case, I think the following is much better but please validate it because I haven't tested it.

```suggestion
type putOptions struct {
	user string
}

type PutOption func(*putOptions)

func newPutOptions(opts []PutOption) *putOptions {
	p := &putOptions{}
	for _, opt := range opts {
		opt(p)
	}

	return p
}
```